Html to pug

BJordi opened the pull request:
## Summary

- Migrated html code to pug in Login and Signup views.

Commit:
Pug
- Installed pug library.
- Installed pug-plain-loader library.

Commit:
Pug
- Migrated html code to pug in Login and Signup views.

FrankIglesias commented on src/views/Login.vue:
```diff
@@ -1,40 +1,21 @@
-  </div>
+<template lang="pug">
+  .container
+    img.logo(src='../assets/wolox-logo.png')
```
add alt

FrankIglesias commented on src/views/Login.vue:
```diff
@@ -1,40 +1,21 @@
+        span.error-label(v-if='!$v.email.required && submitted') Campo requerido
+      .input-container
+        label.label(for='password') Password
+        input#password(type='password', v-model='$v.password.$model', :class="['input', $v.password.$error && ($v.password.required || submitted ) ? 'error-input': '']")
```
indent props

FrankIglesias commented on src/views/Login.vue:
```diff
@@ -1,40 +1,21 @@
+    form(@submit.prevent='onSubmit')
+      .input-container
+        label.label(for='email') Email
+        input#email(v-model='email', :class="['input', ($v.email.$invalid && submitted) || !$v.email.email ? 'error-input' : '']")
```
same here

FrankIglesias commented on src/views/Login.vue:
```diff
@@ -1,40 +1,21 @@
+        span.error-label(v-if='$v.password.$dirty && $v.password.required && !$v.password.hasNumber') El password debe contener al menos un número
+        span.error-label(v-if='$v.password.$dirty && $v.password.required && !$v.password.hasUppercase') El password debe contener al menos una mayúscula
+      button.primary-button Login
+    .horizontal-division
```
use a border instead

FrankIglesias commented on src/views/SignUp.vue:
```diff
@@ -1,49 +1,28 @@
+        span.error-label(v-if='$v.password.$dirty && $v.password.required && !$v.password.hasUppercase') El password debe contener al menos una mayúscula
+      span.error-label(v-if='formError') {{formError}}
+      button.primary-button Sign up
+    .horizontal-division
```
same to this whole file

FrankIglesias commented:
text inputs usually do not have a for prop... so ids can be deleted

Commit:
HTML TO PUG

- Indented props of the different tags to follow our standard.

Commit:
HTML TO PUG

Usually, text inputs don't have a for prop, so ids can be deleted.

Commit:
FIXES IN CLASSES

- Removed unnecesary div used as separator between buttons, as this division didn't have to be with structure but with style.
- Added a border-bottom prop in replace of the deleted div.

Commit:
Added alt prop to images in login and signup forms.

Commit:
Solved conflicts with master branch

mcallegari10 commented on src/scss/commons/components.scss:
```diff
@@ -56,3 +60,8 @@
+.logo {
+  margin: 25px 0;
+  object-fit: contain;
+}
```
Add an EOL

mcallegari10 commented on src/views/Login.vue:
```diff
@@ -1,40 +1,25 @@
+    img.logo(src='../assets/wolox-logo.png', alt='wolox-books-logo')
+    form.bottom-separator(@submit.prevent='onSubmit')
+      .input-container
+        label.label(for='email') Email
```
For plain text, we use pipe notation. For example:

```pug
label(for='email')
  | Email
```

mcallegari10 commented on src/views/Login.vue:
```diff
@@ -1,40 +1,25 @@
-  </div>
+<template lang="pug">
+  .container
+    img.logo(src='../assets/wolox-logo.png', alt='wolox-books-logo')
```
Instead of doing the attribute separation with commas, just leave a space :smile:

BJordi replied:
Great! :wink::+1:

mcallegari10 commented on src/views/Login.vue:
```diff
@@ -1,40 +1,25 @@
+    form.bottom-separator(@submit.prevent='onSubmit')
+      .input-container
+        label.label(for='email') Email
+        input(v-model='email',
```
No need for the comma

mcallegari10 replied:
Also, remember to put the input type :smile:

mcallegari10 commented on src/views/Login.vue:
```diff
@@ -1,40 +1,25 @@
+      .input-container
+        label.label(for='email') Email
+        input(v-model='email',
+          :class="{'input': true, 'error-input': ($v.email.$invalid && submitted) || !$v.email.email}")
```
The correct way to indent should be:

```pug
input(
  v-model='...'
  class='input'
  :class='{ "error-input": ($v.email.$invalid && submitted) || !$v.email.email }'
)
```

Check that you don't need to add to the class bindings the ones that you always want to be apply

BJordi replied:
Ooooh, that's cool! Thank you!

BJordi replied:
Btw, should I use double quotes in `error-input` key? :thinking:

mcallegari10 commented on src/views/Login.vue:
```diff
@@ -1,40 +1,25 @@
+        input(v-model='email',
+          :class="{'input': true, 'error-input': ($v.email.$invalid && submitted) || !$v.email.email}")
+        span.error-label(v-if='!$v.email.email') El email no tiene un formato válido
+        span.error-label(v-if='!$v.email.required && submitted') Campo requerido
```
Use pipe notation

mcallegari10 commented on src/views/Login.vue:
```diff
@@ -1,40 +1,25 @@
+        span.error-label(v-if='!$v.email.email') El email no tiene un formato válido
+        span.error-label(v-if='!$v.email.required && submitted') Campo requerido
+      .input-container
+        label.label(for='password') Password
```
Pipe notation

mcallegari10 commented on src/views/Login.vue:
```diff
@@ -1,40 +1,25 @@
+      .input-container
+        label.label(for='password') Password
+        input(type='password',
+          v-model='$v.password.$model',
```
Check the indentation and commas

mcallegari10 commented on src/views/Login.vue:
```diff
@@ -1,40 +1,25 @@
+        input(type='password',
+          v-model='$v.password.$model',
+          :class="{'input': true, 'error-input': $v.password.$error && ($v.password.required || submitted )}")
+        span.error-label(v-if='!$v.password.required && submitted') Campo requerido
```
Pipe notation

mcallegari10 commented on src/views/Login.vue:
```diff
@@ -1,40 +1,25 @@
+          | El password debe contener al menos un número
+        span.error-label(v-if='$v.password.$dirty && $v.password.required && !$v.password.hasUppercase')
+          | El password debe contener al menos una mayúscula
+      button.primary-button Login
```
Pipe notation

mcallegari10 commented on src/views/Login.vue:
```diff
@@ -1,40 +1,25 @@
+        span.error-label(v-if='$v.password.$dirty && $v.password.required && !$v.password.hasUppercase')
+          | El password debe contener al menos una mayúscula
+      button.primary-button Login
+    router-link.secondary-button(to='/sign-up') Sign up
```
Pipe notation

mcallegari10 commented on src/views/SignUp.vue:
```diff
@@ -1,49 +1,32 @@
-  </div>
+<template lang="pug">
+  .container
+    img.logo(src='../assets/wolox-logo.png', alt='wolox-books-logo')
```
Check the comments of the previous file in this one too :smile:

Commit:
HTML TO PUG: Fixes asked for the code reviewer

- Added EOL.
- In Pug, Use of pipe notation for all the elements that have text.
- In pug, fix indentation of elements with multiple attributes.
- in Pug, removed the commas that separate the elements's attributes.
- In pug, added email type to input.

Commit:
Solved conflicts with master branch
